frontend/js/src/playlists/components/DuplicateTrackModal.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+
+type DuplicateTrackModalProps = {
+  message: string;
+  dontAskAgain: boolean;
+  setDontAskAgain: (value: boolean) => void;
+};
+
+export default NiceModal.create((props: DuplicateTrackModalProps) => {
+  const modal = useModal();
+  const { message, dontAskAgain: initialDontAskAgain, setDontAskAgain } = props;
+
+  // Tried to use dontAskAgain directly, but the checkbox state wasnt updating when toggled.
+  const [localDontAskAgain, setLocalDontAskAgain] = React.useState(
+    initialDontAskAgain
+  );
```

Comment:
Makes sense that it would not change the controlled state, because ticking the checkbox did not call `setDontAskAgain`, it was only called after confirming or cancelling.

There is another way to do this using an uncontrolled checkbox (meaning we don't save its state), but I do not recommend it as I think keeping a local state like this works fine.
For more context read this: https://react.dev/reference/react-dom/components/input#my-checkbox-doesnt-update-when-i-click-on-it

However I think you can remove the comment.